indico/modules/events/api.py

```diff
@@ -492,6 +457,15 @@ def category(self, idlist):
         query = self._update_query(query)
         return self.serialize_events(x for x in query if self._filter_event(x) and x.can_access(self.user))
 
+    def category_extra(self, ids):
+        has_future_events = (False if not self._toDT else
+                             True in [IndexesHolder().getById('categoryDateAll').hasObjectsAfter(cat_id, self._toDT)
```

Comment:
if self._toDT is None:
        has_future_events = False
    else:
        query = Event.find(Event.category_id.in_(ids), ~Event.is_deleted, Event.start_dt > self._toDT)
        has_future_events = db.session.query(query.exists()).one()[0]